py/tools/venv_bin/src/main.rs

```diff
@@ -55,17 +70,63 @@ struct VenvArgs {
     /// activate scripts.
     #[arg(long)]
     venv_name: String,
+
+    /// The mechanism to use in building a virtualenv. Could be static, could be
+    /// dynamic. Allows us to use the same tool statically as dynamically, which
+    /// may or may not be a feature.
+    #[arg(long)]
+    #[clap(default_value = "dynamic-symlink")]
+    mode: VenvMode,
+
+    /// The interpreter version. Must be supplied because there are parts of the
+    /// venv whose path depend on the precise interpreter version. To be sourced from
+    /// PyRuntimeInfo.
+    #[arg(long)]
+    version: Option<String>,
 }
 
 fn venv_cmd_handler(args: VenvArgs) -> miette::Result<()> {
     let pth_file = py::PthFile::new(&args.pth_file, args.pth_entry_prefix);
-    py::create_venv(
-        &args.python,
-        &args.location,
-        Some(pth_file),
-        args.collision_strategy.unwrap_or_default().into(),
-        &args.venv_name,
-    )
+    match args.mode {
+        VenvMode::DynamicSymlink => py::create_venv(
+            &args.python,
+            &args.location,
+            Some(pth_file),
+            args.collision_strategy.unwrap_or_default().into(),
+            &args.venv_name,
+        ),
+        VenvMode::StaticSymlink => {
+            let venv = py::venv::create_empty_venv(
+                &args.python,
+                py::venv::PythonVersionInfo::from_str(&args.version.unwrap())?,
+                &args.location,
+                &args.env_file,
+            )?;
+
+            py::venv::populate_venv_with_copies(
+                venv,
+                pth_file,
+                args.bin_dir.unwrap(),
+                args.collision_strategy.unwrap_or_default().into(),
+            )?;
+
+            Ok(())
+        }
+        VenvMode::StaticPth => {
+            let venv = py::venv::create_empty_venv(
+                &args.python,
+                py::venv::PythonVersionInfo::from_str(&args.version.unwrap())?,
```

Comment:
.unwrap() here sounds dangerous. do we want to check and print an error message?